modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java

```diff
@@ -619,6 +619,8 @@ private void onDiscovery0(DiscoveryNotification notification) {
                     boolean incMinorTopVer;
 
                     if (customMsg instanceof ChangeGlobalStateMessage) {
+                        ctx.state().onStateMessage();
```

Comment:
We don't need this new method in ClusterStateProcessor: it's okay to add this piece of logic right into existing onStateChangeMessage.

Reply:
`onStateChangeMessage` invokes from other code. But we need to invoke `onStateMessage()` only for this case.